pgtricks/pg_dump_splitsort_py3.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+import functools
+
+COPY_RE = re.compile(r'COPY .*? \(.*?\) FROM stdin;\n$')
+
+
+def try_float(s):
+    if not s or s[0] not in '0123456789.-':
+        # optimization
+        return s
+    try:
+        return float(s)
+    except ValueError:
+        return s
+
+
+def linecomp(l1, l2):
+    p1 = l1.split('\t', 1)
+    p2 = l2.split('\t', 1)
+    v1 = try_float(p1[0])
+    v2 = try_float(p2[0])
+    result = (v1 > v2) - (v1 < v2)
+    if not result and len(p1) == len(p2) == 2:
+        return linecomp(p1[1], p2[1])
+    return result
+
+DATA_COMMENT_RE = re.compile('-- Data for Name: (?P<table>.*?); '
+                             'Type: TABLE DATA; '
+                             'Schema: (?P<schema>.*?);')
+SEQUENCE_SET_RE = re.compile(r'-- Name: .+; Type: SEQUENCE SET; Schema: |'
+                             r"SELECT pg_catalog\.setval\('")
+
+class Matcher(object):
+    def __init__(self):
+        self._match = None
+
+    def match(self, pattern, data):
+        self._match = pattern.match(data)
+        return self._match
+
+    def __getattr__(self, attname):
+        if not self._match:
+            raise ValueError('Pattern did not match')
+        return getattr(self._match, attname)
+
+
+def split_sql_file(sql_filepath):
+
+    directory = os.path.dirname(sql_filepath)
+
+    output = None
+    buf = []
+
+    def flush():
+        output.writelines(buf)
+        buf[:] = []
+
+    def new_output(filename):
+        if output:
+            output.close()
+        return open(os.path.join(directory, filename), 'w')
+
+    copy_lines = None
+    counter = 0
+    output = new_output('0000_prologue.sql')
+    matcher = Matcher()
+
+    for line in open(sql_filepath):
+        if copy_lines is None:
+            if line in ('\n', '--\n'):
+                buf.append(line)
+            elif line.startswith('SET search_path = '):
+                flush()
+                buf.append(line)
+            else:
+                if matcher.match(DATA_COMMENT_RE, line):
+                    counter += 1
+                    output = new_output(
+                        '{counter:04}_{schema}.{table}.sql'.format(
+                            counter=counter,
+                            schema=matcher.group('schema'),
+                            table=matcher.group('table')))
+                elif COPY_RE.match(line):
+                    copy_lines = []
+                elif SEQUENCE_SET_RE.match(line):
+                    pass
+                elif 1 <= counter < 9999:
+                    counter = 9999
+                    output = new_output('%04d_epilogue.sql' % counter)
+                buf.append(line)
+                flush()
+        else:
+            if line == '\\.\n':
+                copy_lines.sort(key=functools.cmp_to_key(linecomp))
```

Comment:
list.sort no longer takes cmp-style funcs, and instead takes a "key" function. I opted for using the cmp_to_key helper, as pointed out in the [docs](https://docs.python.org/3/library/stdtypes.html#list.sort)